lib/common/util.js

```diff
@@ -545,3 +545,43 @@ function normalizeArguments(globalContext, localConfig, options) {
 }
 
 util.normalizeArguments = normalizeArguments;
+
+/**
+ * Throttle requests according to a `maxApiCalls` limit.
+ *
+ * @param {function} makeRequestFn - The function that will be called.
+ * @param {object=} options - Configuration object.
+ * @param {number} options.maxApiCalls - The maximum number of API calls to
+ *     make.
+ */
+function iterator(makeRequestFn, options) {
+  var stream = through.obj();
+
+  var requestsMade = 0;
+  var requestsToMake = -1;
+
+  options = options || {};
```

Comment:
Could we add an option for `onreading` or something similar? I feel like we could consolidate all the `stream.on('reading', function() { ... });` lines.